src/format/parser/KeepUnused.ts

```diff
@@ -0,0 +1,72 @@
+import { KeepUnusedConfig } from '../config';
+import ImportNode from './ImportNode';
+
+export default class KeepUnused {
+  private readonly entries: Map<ImportNode, KeepUnusedMatcher[]>;
+
+  constructor(cfg: KeepUnusedConfig[] = [], nodes: ImportNode[]) {
+    const matchers = cfg.map(entry => new KeepUnusedMatcher(entry));

```

Comment:
Say you have entries defined as this:
```ts
private readonly entries_: { path: RegExp, names?: RegExp[] }[];
```
Then the following is all you want:
```suggestion
    this.entries_= cfg
      .filter(e => e.path)
      .map(e => ({path: new RegExp(e.path), names: e.names?.map(n => new RegExp(n))}));
```
And you can remove the `KeepUnusedMatcher` class, or simplify it as an interface.

Reply:
I don't think so, see my comment above why `Map` is necessary. I had a similar approach before and it did not work.